// tailwindcss-classnames/lib/cli/lib/non-configurable/typography.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var fontSmoothing = ['antialiased', 'subpixel-antialiased'];
var fontStyle = ['italic', 'non-italic'];
var fontVariantNumeric = [
    'normal-nums',
    'ordinal',
    'slashed-zero',
    'lining-nums',
    'oldstyle-nums',
    'proportional-nums',
    'tabular-nums',
    'diagonal-fractions',
    'stacked-fractions',
];
var listStylePosition = ['list-inside', 'list-outside'];
var textAlign = ['text-left', 'text-center', 'text-right', 'text-justify'];
var textDecoration = ['underline', 'line-through', 'no-underline'];
var textTransform = ['uppercase', 'lowercase', 'capitalize', 'normal-case'];
var textOverflow = ['truncate', 'text-ellipsis', 'text-clip'];
var verticalAlign = [
    'align-baseline',
    'align-top',
    'align-middle',
    'align-bottom',
    'align-text-top',
    'align-text-bottom',
    'align-sub',
    'align-super',
];
var whitespace = [
    'whitespace-normal',
    'whitespace-nowrap',
    'whitespace-pre',
    'whitespace-pre-line',
    'whitespace-pre-wrap',
];
var wordBreak = ['break-normal', 'break-words', 'break-all'];
var textDecorationStyle = [
    'decoration-solid',
    'decoration-double',
    'decoration-dotted',
    'decoration-dashed',
    'decoration-wavy',
];
exports.default = {
    fontSmoothing: fontSmoothing,
    fontStyle: fontStyle,
    fontVariantNumeric: fontVariantNumeric,
    listStylePosition: listStylePosition,
    textAlign: textAlign,
    textDecoration: textDecoration,
    textTransform: textTransform,
    textOverflow: textOverflow,
    verticalAlign: verticalAlign,
    whitespace: whitespace,
    wordBreak: wordBreak,
    textDecorationStyle: textDecorationStyle,
};
