// tailwindcss-classnames/es/cli/lib/non-configurable/interactivity.js
var appearance = ['appearance-none'];
var pointerEvents = ['pointer-events-none', 'pointer-events-auto'];
var resize = ['resize-none', 'resize', 'resize-y', 'resize-x'];
var userSelect = ['select-none', 'select-text', 'select-all', 'select-auto'];
var scrollSnap = [
    // scroll snap align
    'snap-start',
    'snap-end',
    'snap-center',
    'snap-align-none',
    // scroll snap stop
    'snap-normal',
    'snap-always',
    // scroll snap type
    'snap-none',
    'snap-x',
    'snap-y',
    'snap-both',
    'snap-mandatory',
    'snap-proximity',
];
var scrollBehavior = ['scroll-auto', 'scroll-smooth'];
var touchAction = [
    'touch-auto',
    'touch-none',
    'touch-pan-x',
    'touch-pan-left',
    'touch-pan-right',
    'touch-pan-y',
    'touch-pan-up',
    'touch-pan-down',
    'touch-pinch-zoom',
    'touch-manipulation',
];
export default {
    appearance: appearance,
    pointerEvents: pointerEvents,
    resize: resize,
    userSelect: userSelect,
    scrollSnap: scrollSnap,
    scrollBehavior: scrollBehavior,
    touchAction: touchAction,
};
