// tailwindcss-classnames/es/index.js
/* eslint-disable */
/* tslint:disable */
//////////////////////////////////////////////////////////////////////////////
/// Autogenerated by tailwindcss-classnames CLI. https://git.io/JtsPU
/// DO NOT EDIT THIS FILE DIRECTLY!
//////////////////////////////////////////////////////////////////////////////
import classnamesLib from 'clsx';
//////////// Accessibility Utility functions
export var screenReaders = classnamesLib;
//////////// Backgrounds Utility functions
export var backgroundAttachment = classnamesLib;
export var backgroundClip = classnamesLib;
export var backgroundOrigin = classnamesLib;
export var backgroundRepeat = classnamesLib;
export var backgroundOpacity = classnamesLib;
export var backgroundColor = classnamesLib;
export var backgroundPosition = classnamesLib;
export var backgroundSize = classnamesLib;
export var backgroundImage = classnamesLib;
export var gradientColorStops = classnamesLib;
//////////// Borders Utility functions
export var borderStyle = classnamesLib;
export var divideStyle = classnamesLib;
export var outlineStyle = classnamesLib;
export var borderColor = classnamesLib;
export var borderOpacity = classnamesLib;
export var borderRadius = classnamesLib;
export var borderWidth = classnamesLib;
export var divideColor = classnamesLib;
export var divideOpacity = classnamesLib;
export var divideWidth = classnamesLib;
export var ringColor = classnamesLib;
export var ringWidth = classnamesLib;
export var ringOpacity = classnamesLib;
export var ringOffsetColor = classnamesLib;
export var ringOffsetWidth = classnamesLib;
export var outlineOffset = classnamesLib;
export var outlineWidth = classnamesLib;
export var outlineColor = classnamesLib;
//////////// Tables Utility functions
export var borderCollapse = classnamesLib;
export var tableLayout = classnamesLib;
//////////// Effects Utility functions
export var mixBlendMode = classnamesLib;
export var backgroundBlendMode = classnamesLib;
export var boxShadow = classnamesLib;
export var boxShadowColor = classnamesLib;
export var opacity = classnamesLib;
//////////// TransitionsAndAnimations Utility functions
export var transitionProperty = classnamesLib;
export var transitionDuration = classnamesLib;
export var transitionTimingFunction = classnamesLib;
export var transitionDelay = classnamesLib;
export var animation = classnamesLib;
//////////// Filters Utility functions
export var filter = classnamesLib;
export var backdropFilter = classnamesLib;
export var blur = classnamesLib;
export var brightness = classnamesLib;
export var contrast = classnamesLib;
export var dropShadow = classnamesLib;
export var grayscale = classnamesLib;
export var hueRotate = classnamesLib;
export var invert = classnamesLib;
export var saturate = classnamesLib;
export var sepia = classnamesLib;
export var backdropBlur = classnamesLib;
export var backdropBrightness = classnamesLib;
export var backdropContrast = classnamesLib;
export var backdropGrayscale = classnamesLib;
export var backdropHueRotate = classnamesLib;
export var backdropInvert = classnamesLib;
export var backdropOpacity = classnamesLib;
export var backdropSaturate = classnamesLib;
export var backdropSepia = classnamesLib;
//////////// FlexBox Utility functions
export var flexDirection = classnamesLib;
export var flexWrap = classnamesLib;
export var alignItems = classnamesLib;
export var alignContent = classnamesLib;
export var alignSelf = classnamesLib;
export var placeContent = classnamesLib;
export var placeItems = classnamesLib;
export var placeSelf = classnamesLib;
export var justifyContent = classnamesLib;
export var justifyItems = classnamesLib;
export var justifySelf = classnamesLib;
export var flex = classnamesLib;
export var flexBasis = classnamesLib;
export var flexGrow = classnamesLib;
export var flexShrink = classnamesLib;
export var order = classnamesLib;
//////////// Grid Utility functions
export var gridAutoFlow = classnamesLib;
export var gridTemplateColumns = classnamesLib;
export var gridAutoColumns = classnamesLib;
export var gridColumn = classnamesLib;
export var gridColumnStart = classnamesLib;
export var gridColumnEnd = classnamesLib;
export var gridTemplateRows = classnamesLib;
export var gridAutoRows = classnamesLib;
export var gridRow = classnamesLib;
export var gridRowStart = classnamesLib;
export var gridRowEnd = classnamesLib;
export var gap = classnamesLib;
//////////// Spacing Utility functions
export var padding = classnamesLib;
export var margin = classnamesLib;
export var space = classnamesLib;
//////////// Interactivity Utility functions
export var appearance = classnamesLib;
export var pointerEvents = classnamesLib;
export var resize = classnamesLib;
export var userSelect = classnamesLib;
export var scrollSnap = classnamesLib;
export var scrollBehavior = classnamesLib;
export var touchAction = classnamesLib;
export var cursor = classnamesLib;
export var caretColor = classnamesLib;
export var willChange = classnamesLib;
export var accentColor = classnamesLib;
export var scrollPadding = classnamesLib;
export var scrollMargin = classnamesLib;
//////////// Layout Utility functions
export var display = classnamesLib;
export var boxDecorationBreak = classnamesLib;
export var boxSizing = classnamesLib;
export var container = classnamesLib;
export var float = classnamesLib;
export var clear = classnamesLib;
export var isolation = classnamesLib;
export var objectFit = classnamesLib;
export var overflow = classnamesLib;
export var overscrollBehavior = classnamesLib;
export var position = classnamesLib;
export var visibility = classnamesLib;
export var breakBefore = classnamesLib;
export var breakInside = classnamesLib;
export var breakAfter = classnamesLib;
export var objectPosition = classnamesLib;
export var inset = classnamesLib;
export var zIndex = classnamesLib;
export var aspectRatio = classnamesLib;
export var columns = classnamesLib;
//////////// Sizing Utility functions
export var width = classnamesLib;
export var minWidth = classnamesLib;
export var maxWidth = classnamesLib;
export var height = classnamesLib;
export var minHeight = classnamesLib;
export var maxHeight = classnamesLib;
//////////// SVG Utility functions
export var fill = classnamesLib;
export var stroke = classnamesLib;
export var strokeWidth = classnamesLib;
//////////// Transforms Utility functions
export var hardwareAcceleration = classnamesLib;
export var scale = classnamesLib;
export var rotate = classnamesLib;
export var translate = classnamesLib;
export var skew = classnamesLib;
export var transformOrigin = classnamesLib;
//////////// Typography Utility functions
export var fontSmoothing = classnamesLib;
export var fontStyle = classnamesLib;
export var fontVariantNumeric = classnamesLib;
export var listStylePosition = classnamesLib;
export var textAlign = classnamesLib;
export var textDecoration = classnamesLib;
export var textTransform = classnamesLib;
export var textOverflow = classnamesLib;
export var verticalAlign = classnamesLib;
export var whitespace = classnamesLib;
export var wordBreak = classnamesLib;
export var textDecorationStyle = classnamesLib;
export var fontFamily = classnamesLib;
export var fontSize = classnamesLib;
export var fontWeight = classnamesLib;
export var letterSpacing = classnamesLib;
export var lineHeight = classnamesLib;
export var listStyleType = classnamesLib;
export var placeholderColor = classnamesLib;
export var placeholderOpacity = classnamesLib;
export var textColor = classnamesLib;
export var textOpacity = classnamesLib;
export var content = classnamesLib;
export var textIndent = classnamesLib;
export var textDecorationColor = classnamesLib;
export var textDecorationThickness = classnamesLib;
export var textUnderlineOffset = classnamesLib;
export var TW = {
    screenReaders: screenReaders,
    backgroundAttachment: backgroundAttachment,
    backgroundClip: backgroundClip,
    backgroundOrigin: backgroundOrigin,
    backgroundRepeat: backgroundRepeat,
    backgroundOpacity: backgroundOpacity,
    backgroundColor: backgroundColor,
    backgroundPosition: backgroundPosition,
    backgroundSize: backgroundSize,
    backgroundImage: backgroundImage,
    gradientColorStops: gradientColorStops,
    borderStyle: borderStyle,
    divideStyle: divideStyle,
    outlineStyle: outlineStyle,
    borderColor: borderColor,
    borderOpacity: borderOpacity,
    borderRadius: borderRadius,
    borderWidth: borderWidth,
    divideColor: divideColor,
    divideOpacity: divideOpacity,
    divideWidth: divideWidth,
    ringColor: ringColor,
    ringWidth: ringWidth,
    ringOpacity: ringOpacity,
    ringOffsetColor: ringOffsetColor,
    ringOffsetWidth: ringOffsetWidth,
    outlineOffset: outlineOffset,
    outlineWidth: outlineWidth,
    outlineColor: outlineColor,
    borderCollapse: borderCollapse,
    tableLayout: tableLayout,
    mixBlendMode: mixBlendMode,
    backgroundBlendMode: backgroundBlendMode,
    boxShadow: boxShadow,
    boxShadowColor: boxShadowColor,
    opacity: opacity,
    transitionProperty: transitionProperty,
    transitionDuration: transitionDuration,
    transitionTimingFunction: transitionTimingFunction,
    transitionDelay: transitionDelay,
    animation: animation,
    filter: filter,
    backdropFilter: backdropFilter,
    blur: blur,
    brightness: brightness,
    contrast: contrast,
    dropShadow: dropShadow,
    grayscale: grayscale,
    hueRotate: hueRotate,
    invert: invert,
    saturate: saturate,
    sepia: sepia,
    backdropBlur: backdropBlur,
    backdropBrightness: backdropBrightness,
    backdropContrast: backdropContrast,
    backdropGrayscale: backdropGrayscale,
    backdropHueRotate: backdropHueRotate,
    backdropInvert: backdropInvert,
    backdropOpacity: backdropOpacity,
    backdropSaturate: backdropSaturate,
    backdropSepia: backdropSepia,
    flexDirection: flexDirection,
    flexWrap: flexWrap,
    alignItems: alignItems,
    alignContent: alignContent,
    alignSelf: alignSelf,
    placeContent: placeContent,
    placeItems: placeItems,
    placeSelf: placeSelf,
    justifyContent: justifyContent,
    justifyItems: justifyItems,
    justifySelf: justifySelf,
    flex: flex,
    flexBasis: flexBasis,
    flexGrow: flexGrow,
    flexShrink: flexShrink,
    order: order,
    gridAutoFlow: gridAutoFlow,
    gridTemplateColumns: gridTemplateColumns,
    gridAutoColumns: gridAutoColumns,
    gridColumn: gridColumn,
    gridColumnStart: gridColumnStart,
    gridColumnEnd: gridColumnEnd,
    gridTemplateRows: gridTemplateRows,
    gridAutoRows: gridAutoRows,
    gridRow: gridRow,
    gridRowStart: gridRowStart,
    gridRowEnd: gridRowEnd,
    gap: gap,
    padding: padding,
    margin: margin,
    space: space,
    appearance: appearance,
    pointerEvents: pointerEvents,
    resize: resize,
    userSelect: userSelect,
    scrollSnap: scrollSnap,
    scrollBehavior: scrollBehavior,
    touchAction: touchAction,
    cursor: cursor,
    caretColor: caretColor,
    willChange: willChange,
    accentColor: accentColor,
    scrollPadding: scrollPadding,
    scrollMargin: scrollMargin,
    display: display,
    boxDecorationBreak: boxDecorationBreak,
    boxSizing: boxSizing,
    container: container,
    float: float,
    clear: clear,
    isolation: isolation,
    objectFit: objectFit,
    overflow: overflow,
    overscrollBehavior: overscrollBehavior,
    position: position,
    visibility: visibility,
    breakBefore: breakBefore,
    breakInside: breakInside,
    breakAfter: breakAfter,
    objectPosition: objectPosition,
    inset: inset,
    zIndex: zIndex,
    aspectRatio: aspectRatio,
    columns: columns,
    width: width,
    minWidth: minWidth,
    maxWidth: maxWidth,
    height: height,
    minHeight: minHeight,
    maxHeight: maxHeight,
    fill: fill,
    stroke: stroke,
    strokeWidth: strokeWidth,
    hardwareAcceleration: hardwareAcceleration,
    scale: scale,
    rotate: rotate,
    translate: translate,
    skew: skew,
    transformOrigin: transformOrigin,
    fontSmoothing: fontSmoothing,
    fontStyle: fontStyle,
    fontVariantNumeric: fontVariantNumeric,
    listStylePosition: listStylePosition,
    textAlign: textAlign,
    textDecoration: textDecoration,
    textTransform: textTransform,
    textOverflow: textOverflow,
    verticalAlign: verticalAlign,
    whitespace: whitespace,
    wordBreak: wordBreak,
    textDecorationStyle: textDecorationStyle,
    fontFamily: fontFamily,
    fontSize: fontSize,
    fontWeight: fontWeight,
    letterSpacing: letterSpacing,
    lineHeight: lineHeight,
    listStyleType: listStyleType,
    placeholderColor: placeholderColor,
    placeholderOpacity: placeholderOpacity,
    textColor: textColor,
    textOpacity: textOpacity,
    content: content,
    textIndent: textIndent,
    textDecorationColor: textDecorationColor,
    textDecorationThickness: textDecorationThickness,
    textUnderlineOffset: textUnderlineOffset,
};
export var classnames = classnamesLib;
export default classnames;
